main.tf

```diff
@@ -44,6 +48,38 @@ resource "azurerm_kubernetes_cluster" "this" {
       identity_ids = var.identity_ids
     }
   }
+  dynamic "key_vault_secrets_provider" {
+    for_each = var.key_vault_secrets_provider_enabled ? ["key_vault_secrets_provider"] : []
+
+    content {
+      secret_rotation_enabled  =true
+    }
+  }
+
+    dynamic "monitor_metrics" {
+  
+    for_each = var.monitor_metrics != null ? [var.monitor_metrics] : []
+
+    content {
+      annotations_allowed = var.monitor_metrics.annotations_allowed
+      labels_allowed      = var.monitor_metrics.labels_allowed
+    }
+  }
+  network_profile {
+    network_plugin      = "azure"
+    load_balancer_sku   = "standard"
+    network_plugin_mode = "overlay"
+    network_policy      = "calico"
+    outbound_type       = "userAssignedNATGateway"
+  }
+ dynamic "oms_agent" {
```

Comment:
This change looks correct, however it does not reflect the title of the PR.

Please I suggest 2 options:

* Split this PR in multiple PRs with smaller changes

OR

* Change the title of this PR and describe all the changes that you are proposing.

Reply:
Thanks for reviewing @zioproto  i have moved monitoring and networking to a separate PR. This will only have the node pool settings and availability zones